backend/handlers.go

```diff
@@ -58,35 +56,45 @@ func (cs *server) publishHandler(w http.ResponseWriter, r *http.Request, _ httpr
 		return
 	}
 	defer body.Close()
-
 	// Unmarshal the JSON message into a map
-	var data map[string]interface{}
-	if err := json.Unmarshal(msg, &data); err != nil {
+	decoder := json.NewDecoder(bytes.NewReader(msg))
+	// Create a new map to store the sensorDataMap
+	var sensorIDMap map[string]interface{}
```

Comment:
You can also try a `map[string][]byte` or `map[string]string` here so you don't have to cast the type later.